### PR TITLE
Add deploy environment

### PR DESCRIPTION
**Pull request by victor77dev**
### Description
- Automatic deployment triggers from branches like, `release/*`, `main`, `production/*`, `lab-*`

### QA
- [ ] Chrome

### Issuer
- https://github.com/Tremendous-Tiny-Team/ttt-web/issues/TICKET_ID

### TODO
- [ ] Security/password check for non-production envs
- [ ] Home page to link to different envs

### CODE REVIEW

File: scripts/deploy.mjs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { exec } from 'child_process';
+import * as ghPages from 'gh-pages';
+
+const branch = process.env.GITHUB_REF.replace('refs/heads/', '');
+const deployEnv = branch.match(
+  /^main|^release|^production|^hotfix|^lab-\d+/
+)?.[0];
```
Comment (victor77dev) [inline scripts/deploy.mjs:7]: @kavimaluskam I found I don't like Prettier that much (or current setting of Prettier).
E.g. I found this kind of breaking lines are hard to read.

File: scripts/deploy.mjs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      user: {
+        name: process.env.GITHUB_ACTOR,
+        email: process.env.GITHUB_ACTOR_EMAIL,
+      },
+      repo: `https://x-access-token:${process.env.GITHUB_TOKEN}@github.com/${process.env.GITHUB_REPOSITORY}.git`,
+      branch: 'gh-pages',
+      dest: deployEnv,
+      message: `Build from ${branch} ${process.env.GITHUB_SHA}`,
+    },
+    (error) => {
```
Comment (victor77dev) [inline scripts/deploy.mjs:34]: Also, this break (maybe). But this one is still ok.

**Review by victor77dev — COMMENTED** [review]
@kavimaluskam some comments about using Prettier

File: scripts/deploy.mjs (hunk 1 above)
Comment (kavimaluskam) [reply scripts/deploy.mjs:7]: Of course i am used to the breaking of prettier already... shall we research on the configuration of prettier later?

I think one advantage of prettier is enforcing us to use the same styling during development so commit / fix about styling is avoided.

Comment (victor77dev) [reply scripts/deploy.mjs:7]: I agree there is advantage of this.
But this `?.[0]` is broken into another line, which I still strange.

**Comment by victor77dev** [comment]
@kavimaluskam I'm force pushing the branch again as this PR shouldn't include those commits.
If you want to try it, try to create another lab branch base on this and merge your changes, say `lab-3/test-xxx`

**Review by kavimaluskam — APPROVED** [review]
I can see from github home page that it's `github-action` bot publishing the version to lab enviornment, is that expected behavior?

**Comment by victor77dev** [comment]
> I can see from github home page that it's `github-action` bot publishing the version to lab enviornment, is that expected behavior?

Yes and no. No for I didn't expect that, but Yes for it looks like it couldn't get the right info for merge commit.

**Pull request merged**